std/json.zig

```diff
@@ -0,0 +1,1308 @@
+                },
+            },
+
+            State.FalseLiteral1 => switch (c) {
+                'a' => p.state = State.FalseLiteral2,
+                else => return error.InvalidLiteral,
+            },
+
+            State.FalseLiteral2 => switch (c) {
+                'l' => p.state = State.FalseLiteral3,
+                else => return error.InvalidLiteral,
+            },
+
+            State.FalseLiteral3 => switch (c) {
+                's' => p.state = State.FalseLiteral4,
+                else => return error.InvalidLiteral,
+            },
+
+            State.FalseLiteral4 => switch (c) {
+                'e' => {
+                    p.state = p.after_value_state;
+                    p.complete = p.state == State.TopLevelEnd;
+                    *token = Token.init(TokenId.False, p.count + 1, 1);
+                },
+                else => {
+                    return error.InvalidLiteral;
+                },
+            },
+
+            State.NullLiteral1 => switch (c) {
+                'u' => p.state = State.NullLiteral2,
+                else => return error.InvalidLiteral,
+            },
+
+            State.NullLiteral2 => switch (c) {
+                'l' => p.state = State.NullLiteral3,
+                else => return error.InvalidLiteral,
+            },
+
+            State.NullLiteral3 => switch (c) {
+                'l' => {
+                    p.state = p.after_value_state;
+                    p.complete = p.state == State.TopLevelEnd;
+                    *token = Token.init(TokenId.Null, p.count + 1, 1);
+                },
+                else => {
+                    return error.InvalidLiteral;
+                },
+            },
+        }
+
+        return false;
+    }
+};
+
+// Validate a JSON string. This does not limit number precision so a decoder may not necessarily
+// be able to decode the string even if this returns true.
+pub fn validate(s: []const u8) bool {
+    var p = StreamingJsonParser.init();
+
+    for (s) |c, i| {
+        var token1: ?Token = undefined;
+        var token2: ?Token = undefined;
+
+        p.feed(c, &token1, &token2) catch |err| {
+            return false;
+        };
+    }
+
+    return p.complete;
+}
+
+const Allocator = std.mem.Allocator;
+const ArenaAllocator = std.heap.ArenaAllocator;
+const ArrayList = std.ArrayList;
+const HashMap = std.HashMap;
+
+pub const ValueTree = struct {
+    arena: ArenaAllocator,
+    root: Value,
+
+    pub fn deinit(self: &ValueTree) void {
+        self.arena.deinit();
+    }
+};
+
+pub const ObjectMap = HashMap([]const u8, Value, mem.hash_slice_u8, mem.eql_slice_u8);
+
+pub const Value = union(enum) {
+    Null,
+    Bool: bool,
+    Integer: i64,
+    Float: f64,
+    String: []const u8,
+    Array: ArrayList(Value),
+    Object: ObjectMap,
+
+    pub fn toString(self: &const Value) void {
+        switch (*self) {
+            Value.Null => {
+                std.debug.warn("null");
```

Comment:
Forgot these were still `std.debug.warn` outputs. Custom formatting functions would be useful here instead.

Reply:
If this function is for debugging - there's a pattern of using `foo.dump()` as a method that returns `void` and prints a bunch of stuff to stderr about `foo` to aid in debugging.